### ISSUE TITLE
Error: Call to a member function find() on a non-object 

### ISSUE THREAD

**Issue by plalanne** [CakeDC/users#393]
Hi,
after installing your plugin i have the error message

> Error: Call to a member function find() on a non-object 

due to a request in the beforRender function.
>public function beforeRender(Event $event)
    {
        $this->loadModel('Projects');
        $projects_list = $this->Projects->find('list')->toArray();
        $this->set('projects_list', $projects_list);
        if (!array_key_exists('_serialize', $this->viewVars) &&
            in_array($this->response->type(), ['application/json', 'application/xml'])
        ) {
            $this->set('_serialize', true);
        }
    }

if i delete the request all is ok.

Of course there is no problem with this request without your plugin

An idea ?

Thanks

**Comment by steinkel** [comment]
Please could you paste the full stack trace?
Debug $this->Projects, looks like this could be the issue?
Thanks,

**Comment by plalanne** [comment]
The result of debug($this->Projects) just before the line where the problem is
`$this->loadModel('Projects');
debug($this->Projects);die();        
$projects_list = $this->Projects->find('list')->toArray();`

is 

` /src/Controller/AppController.php (line 58)
false
`

**Comment by steinkel** [comment]
So looks like the problem is the table is not being loaded correctly, could you debug this issue?

Are you overriding loadModel() in your AppController class?
Debug TableRegistry::get('Projects'); 

Thanks,

**Comment by plalanne** [comment]
The result is 
` /src/Controller/AppController.php (line 61)

object(App\Model\Table\ProjectsTable) {

	'registryAlias' => 'Projects',
	'table' => 'projects',
	'alias' => 'Projects',
	'entityClass' => 'App\Model\Entity\Project',
	'associations' => [
		(int) 0 => 'categories',
		(int) 1 => 'tags'
	],
	'behaviors' => [
		(int) 0 => 'Timestamp'
	],
	'defaultConnection' => 'default',
	'connectionName' => 'default'

}

`

but remember that when i remove the cakedc/users component the problem doesn't exist

**Comment by steinkel** [comment]
what version of the plugin are you using?

**Comment by plalanne** [comment]
in my composer.json i have "cakedc/users": "^3.2"

when i tried the same installation on a new cakephp project the component works well

**Comment by steinkel** [comment]
Please do "composer update" as you could be using an old version of the plugin

**Comment by plalanne** [comment]
i have now the cakedc/users (3.2.2) but there is always the same problem

**Comment by steinkel** [comment]
Could you bin the code for your AppController file? Something happens in loadModel returning false...

**Comment by plalanne** [comment]
`<?php

namespace App\Controller;

use Cake\Controller\Controller;
use Cake\Event\Event;
use Cake\Routing\Router;

class AppController extends Controller
{

    public function initialize()
    {
        parent::initialize();

        $this->loadComponent('RequestHandler');
        $this->loadComponent('Flash');
        $this->loadComponent('CakeDC/Users.UsersAuth');
    }

    public function beforeRender(Event $event)
    {
        $this->loadModel('Projects');
        //Liste des projets pour le menu déroulant de la page d'accueil
        $projects_list = $this->Projects->find('list')->toArray();
        $this->set('projects_list', $projects_list);

        if ($this->request->param('controller') == 'Articles' && $this->request->param('action') == 'contents') {
            $id_project = $this->request['pass'][0];
            $title = $projects_list[$id_project];
        } else {
            $title = "La Documentation de vos applications";
        }
        $this->set('title', $title);

        //Changement de layout pour la partie admin du site
        if (isset($this->request->params['prefix']) && ($this->request->params['prefix'] == 'member' || $this->request->params['prefix'] == 'admin')) {
            $this->set('BASE_URL', Router::URL('/'));
            if($this->request->is('ajax')) $this->viewBuilder()->layout('ajax');
            else $this->viewBuilder()->layout('layout_admin_LTE');
        }

        if (!array_key_exists('_serialize', $this->viewVars) &&
            in_array($this->response->type(), ['application/json', 'application/xml'])
        ) {
            $this->set('_serialize', true);
        }
    }
}
`

**Comment by steinkel** [comment]
Thank you I'll need some time to reproduce the issue locally, I'll let you know if any progress later today/tomorrow.

**Comment by plalanne** [comment]
Hi,
i look at the code of the plugin and i don't understand why you override the loadmodel method in the UsersController.php

**Comment by eniacom** [comment]
Hi, we have the same problem...
The solution is to invoke loadModel as defined in cakeFramework.
Override loadModel to force specific model is not a good practice :(
You should define a new funcition i.e. MyLoadModel instead of loadModel and invoke MyLoadModel() when you should load UsersTable Object according to custom configure users table.
```
public function MyLoadModel()
{
     return parent::loadModel(Configure::read('Users.table'));
}
```
Sorry for my English

**Comment by plalanne** [comment]
Hi, Thanks for your answer. 
Ok i replace the loadmodel function in the userscontroller but after where the modification have to be done ?

**Comment by vnt** [comment]
Help!!!!
I have the same problem?

**Comment by eniacom** [comment]
If you have change the loadModel function, now you should replace all calls in plugin code to your MyLoadModel() function instead of old loadModel() function:
`$this->MyLoadModel()`
instead of
`$this->loadModel()`
This replacement it should be done from CakeDC mantainers developer :)
This is my idea.
I don't know if is the best practice to solve this bug.

**Comment by ajibarra** [comment]
Closed by #418
